tooling/backend_interface/src/lib.rs

```diff
@@ -104,6 +106,15 @@ impl Backend {
     fn crs_directory(&self) -> PathBuf {
         self.backend_directory().join("crs")
     }
+
+    fn assert_correct_version(&self) -> Result<String, BackendError> {
+        let binary_path = self.binary_path();
+        let version_string = VersionCommand{}.run(binary_path)?;
+        if version_string.as_str() != BB_VERSION {
+            println!("WARNING!: Configured backend version `{:}` is different from expected `{:}`", version_string, BB_VERSION);
+        }
+        Ok(version_string)
```

Comment:
We should make sure this doesn't run for backends other than the default bb install. i.e. not if `NARGO_BACKEND_PATH` is set or the name doesn't match `acvm-backend-barretenberg.

Reply:
hmm, this is a hack for specific backend, even for this backend it's not a good check because I could configure backend-barretenebrg with different version deliberately, forget about it and later be surprised this is not working. Also one can name their backend something else than `acvm-backend-barretenberg`, therefore value of this is limited.

Reply:
Added checking only when installation path contains `barretenberg` in
9ca358d: chore: assert version on ly on barretenberg backends